src/Neo.CLI/CLI/MainService.Tools.cs

```diff
@@ -445,6 +571,214 @@ private string ClearHexString(string hexString)
+        }
+
+        private string ScriptsToOpCode(List<byte> scripts)
+        {
+            //Initialize all OpCodes
+            var OperandSizePrefixTable = new int[256];
+            var OperandSizeTable = new int[256];
+            foreach (FieldInfo field in typeof(OpCode).GetFields(BindingFlags.Public | BindingFlags.Static))
+            {
+                var attribute = field.GetCustomAttribute<OperandSizeAttribute>();
+                if (attribute == null) continue;
+                int index = (int)(OpCode)field.GetValue(null);
+                OperandSizePrefixTable[index] = attribute.SizePrefix;
+                OperandSizeTable[index] = attribute.Size;
+            }
+            //Initialize all InteropService
+            var dic = new Dictionary<uint, string>();
+            ApplicationEngine.Services.ToList().ForEach(p => dic.Add(p.Value.Hash, p.Value.Name));
+
+            //Analyzing Scripts
+            var result = new List<string>();
+            while (scripts.Count > 0)
+            {
+                var op = (OpCode)scripts[0];
+                var operandSizePrefix = OperandSizePrefixTable[scripts[0]];
+                var operandSize = OperandSizeTable[scripts[0]];
+                scripts.RemoveAt(0);
+
+                var onlyOpCode = true;
+                if (operandSize > 0)
+                {
+                    var operand = scripts.Take(operandSize).ToArray();
+                    if (op.ToString().StartsWith("PUSHINT"))
+                    {
+                        result.Add($"{op} {new BigInteger(operand)}");
+                    }
+                    else if (op == OpCode.SYSCALL)
+                    {
+                        result.Add($"{op} {dic[BitConverter.ToUInt32(operand)]}");
+                    }
+                    else
+                    {
+                        result.Add($"{op} {operand.ToHexString()}");
+                    }
+                    scripts.RemoveRange(0, operandSize);
+                    onlyOpCode = false;
+                }
+                if (operandSizePrefix > 0)
+                {
+                    var bytes = scripts.Take(operandSizePrefix).ToArray();
+                    var number = bytes.Length == 1 ? bytes[0] : (int)new BigInteger(bytes);
+                    scripts.RemoveRange(0, operandSizePrefix);
+                    var operand = scripts.Take(number).ToArray();
+
+                    var asicii = Encoding.Default.GetString(operand);
+                    asicii = asicii.Any(p => p < '0' || p > 'z') ? operand.ToHexString() : asicii;
+
+                    result.Add($"{op} {(number == 20 ? new UInt160(operand).ToString() : asicii)}");
+                    scripts.RemoveRange(0, number);
+                    onlyOpCode = false;
+                }
+                if (onlyOpCode)
+                {
+                    result.Add($"{op}");
+                }
+            }
+            return "\r\n" + string.Join("\r\n", result.ToArray());
```

Comment:
Use `Environment.NewLine` we do support `Operating Systems`.

Reply:
Already modified